operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -172,10 +172,16 @@ private String getLockName(String namespace, String name) {
                 log.debug("{}: Lock {} acquired", reconciliation, lockName);
                 Lock lock = res.result();
                 try {
-                    Future<T> result = callable.call();
-                    lock.release();
-                    log.debug("{}: Lock {} released", reconciliation, lockName);
-                    result.setHandler(handler);
+                    callable.call().setHandler(callableRes -> {
+                        if (callableRes.succeeded()) {
+                            handler.complete(callableRes.result());
+                        } else {
+                            handler.fail(callableRes.cause());
+                        }
+
+                        log.debug("{}: Lock {} released", reconciliation, lockName);
+                        lock.release();
```

Comment:
we should use the same order of release and log as before (as it happens in the catch as well) to log that it's actually released.

Reply:
That makes sense. Fixed.